middlewares/replacePath_test.go

```diff
@@ -0,0 +1,42 @@
+package middlewares_test
+
+import (
+	"github.com/containous/traefik/middlewares"
+	"net/http"
+	"testing"
+)
+
+func TestReplacePath(t *testing.T) {
+	const replacementPath = "/replacement-path"
+
+	handler := &middlewares.ReplacePath{
+		Path: replacementPath,
+	}
+
+	var newPath, oldPath string
+	handler.SetHandler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		newPath = r.URL.Path
+		oldPath = r.Header.Get("X-Replaced-Path")
+	}))
+
+	paths := []string{
+		"/example",
+		"/some/really/long/path",
+	}
+
+	for _, path := range paths {
+		req, err := http.NewRequest("ANY", "http://localhost"+path, nil)
```

Comment:
Unless `ANY` is some kind of special place holder accepted by Go/HTTP, I'd rather like to use `GET`. This will be safe to work in the future too.